src/Raven.Server/Documents/Handlers/Debugging/ServerWideDebugInfoPackageHandler.cs

```diff
@@ -230,63 +218,57 @@ public async Task GetInfoPackage()
             }
         }
 
-        private static void DumpStackTraces(ZipArchive archive, string prefix)
+        private static async Task WriteLogFile(ZipArchive archive)
         {
-            var zipArchiveEntry = archive.CreateEntry($"{prefix}/stacktraces.json", CompressionLevel.Optimal);
+            var prefix = $"{_serverWidePrefix}/{DateTime.UtcNow:yyyy-MM-dd H:mm:ss}.txt";
 
-            var threadsUsage = new ThreadsUsage();
-            var sp = Stopwatch.StartNew();
-
-            using (var stackTraceStream = zipArchiveEntry.Open())
-            using (var sw = new StringWriter())
+            try
             {
-                try
-                {
-                    if (Debugger.IsAttached)
-                        throw new InvalidOperationException("Cannot get stack traces when debugger is attached");
+                var ms = new MemoryStream();
+                var isInfoEnabled = LoggingSource.Instance.IsInfoEnabled;
+                var isOperationsEnabled = LoggingSource.Instance.IsOperationsEnabled;
 
-                    ThreadsHandler.OutputResultToStream(sw);
+                if (isInfoEnabled == false)
+                    LoggingSource.Instance.IsInfoEnabled = true;
+                if (isOperationsEnabled == false)
+                    LoggingSource.Instance.IsOperationsEnabled = true;
 
-                    var result = JObject.Parse(sw.GetStringBuilder().ToString());
+                LoggingSource.Instance.AttachPipeSink(ms);
+                var sp = Stopwatch.StartNew();
 
-                    var wait = 100 - sp.ElapsedMilliseconds;
-                    if (wait > 0)
-                    {
-                        // I expect this to be _rare_, but we need to wait to get a correct measure of the cpu
-                        Thread.Sleep((int)wait);
-                    }
+                while (sp.ElapsedMilliseconds < 15000)
```

Comment:
why not simply `await Task.Delay(15_000);`?

Reply:
done